plugins/modules/packaging/os/yum_versionlock.py

```diff
@@ -0,0 +1,150 @@
+short_description: Locks / Unlocks an installed package(s) from being updated by yum package manager.
+description:
+     - This module adds installed packages to yum versionlock to prevent the package from being updated.
+options:
+  name:
+    description:
+      - Package name or a list of packages.
+    type: list
+    required: true
+    elements: str
+  state:
+    description:
+    - If state is C(present) or C(locked), package(s) will be added to yum versionlock list.
+    - If state is C(absent) or C(unlocked), package(s) will be removed from yum versionlock list.
+    choices: [ 'absent', 'locked', 'present', 'unlocked' ]
+    type: str
+    default: locked
+notes:
+    - Requires yum-plugin-versionlock package on the remote node.
+    - Supports C(check_mode).
+requirements:
+- yum
+- yum-versionlock
+author:
+    - Florian Paul Hoberg (@florianpaulhoberg)
+'''
+
+EXAMPLES = r'''
+- name: Prevent Apache / httpd from being updated
+  community.general.yum_versionlock:
+    state: locked
+    name: httpd
+
+- name: Prevent multiple packages from being updated
+  community.general.yum_versionlock:
+    state: locked
+    name:
+    - httpd
+    - nginx
+    - haproxy
+    - curl
+
+- name: Unlock Apache / httpd to be updated again
+  community.general.yum_versionlock:
+    state: unlocked
+    package: httpd
+'''
+
+RETURN = r'''
+packages:
+    description: A list of package(s) in versionlock list.
+    returned: success
+    type: list
+    elements: str
+    sample: [ 'httpd' ]
+state:
+    description: State of package(s).
+    returned: success
+    type: str
+    sample: locked
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+
+
+class YumVersionLock:
+    def __init__(self, module):
+        self.module = module
+        self.params = module.params
+        self.yum_bin = module.get_bin_path('yum', required=True)
+
+    def get_versionlock_packages(self):
+        """ Get an overview of all packages on yum versionlock """
+        rc, out, err = self.module.run_command([self.yum_bin, "versionlock", "list"])
+        if rc == 0:
+            return out
+        elif rc == 1 and 'o such command:' in err:
+            self.module.fail_json(msg="Error: Please install rpm package yum-plugin-versionlock : " + to_native(err) + to_native(out))
+        self.module.fail_json(msg="Error: " + to_native(err) + to_native(out))
+
+    def ensure_state(self, package, command):
+        """ Ensure package state """
+        rc, out, err = self.module.run_command([self.yum_bin, "-q", "versionlock", command, package])
+        if rc == 0:
+            return True
+        self.module.fail_json(msg="Error: " + to_native(err) + to_native(out))
+
+
+def main():
+    """ start main program to add/remove a package to yum versionlock"""
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(default='locked', choices=['present', 'locked', 'unlocked', 'absent']),
```

Comment:
I think `present` and `absent` are not very useful and only they only create confusion. Is it OK if I remove them and only leave `locked` and `unlocked`?

Reply:
I guess just having one set of values is totally fine. Not sure whether `present`/`absent` or `locked`/`unlocked` is the better choice though. The module name is `yum_versionlock`, so you could argue that you want a version lock for some package present or absent.

Reply:
You're right. `present`/`absent` makes more sense, but maybe `locked`/`unlocked` would be easier for users to use.

Reply:
> The module name is `yum_versionlock`, so you could argue that you want a version lock for some package present or absent.

agreed, `present / absent` sound consistently with the module's name (imo)

Reply:
@felixfontein @Andersson007 thanks for your suggestions.
I removed `locked` and `unlocked` in https://github.com/ansible-collections/community.general/pull/1405/commits/cc38724791d7f8d900b73da7ed9e71cf4dc5463d.